Framework/Algorithms/test/RebinTest.h

```diff
@@ -48,6 +48,70 @@ class RebinTest : public CxxTest::TestSuite {
     NUMBINS = 50;
   }
 
+  /* some basic validation tests */
+
+  void test_make_three_params_default() {
+    // test that if only a binwidth is given, will create a full set of rebin params
+    Workspace2D_sptr test_in1D = Create1DWorkspace(50);
+    AnalysisDataService::Instance().add("test_in1D", test_in1D);
+
+    Rebin rebin;
+    rebin.initialize();
+    rebin.setPropertyValue("InputWorkspace", "test_in1D");
+    rebin.setPropertyValue("OutputWorkspace", "test_out");
+    rebin.setPropertyValue("Params", "2.0");
+    // verify that with the single rebin param, the algo will successfuly execute
+    TS_ASSERT_THROWS_NOTHING(rebin.execute());
+    TS_ASSERT(rebin.isExecuted());
+
+    // ensure that the properties are not changed in default mode
+    // the "Params" property should still have a single value, 2.0
+    std::vector<double> rbParams = rebin.getProperty("Params");
+    TS_ASSERT_EQUALS(rbParams.size(), 1);
+    TS_ASSERT_EQUALS(2.0, rbParams[0]);
+
+    AnalysisDataService::Instance().remove("test_in1D");
+    AnalysisDataService::Instance().remove("test_out");
+  }
+
+  void test_failure_bad_log_binning_endpoints() {
+    // ensure that rebinning will fail if log binning from negative to positive
+    // this fails fast, due to the rebin param property validator
+    Rebin rebin;
+    rebin.initialize();
+    TS_ASSERT_THROWS_ANYTHING(rebin.setPropertyValue("Params", "-1.0,-1.0,1000.0"));
+    TS_ASSERT(!rebin.isExecuted());
+  }
+
+  void test_failure_bad_rebin_params() {
+    // ensure that rebinning will fail if log binning from negative to positive
+    // this fails fast, due to the rebin param property validator
+    Rebin rebin;
+    rebin.initialize();
+    TS_ASSERT_THROWS_ANYTHING(rebin.setPropertyValue("Params", "-1.0,-1.0,10.0"));
```

Comment:
Can this be done with initializer lists? 
```
    TS_ASSERT_THROWS_ANYTHING(rebin.setProperty("Params", {-1.0,-1.0,10.0}));
```
or
```
    TS_ASSERT_THROWS_ANYTHING(rebin.setProperty("Params", std::vector<double>{-1.0,-1.0,10.0}));
```
?

Reply:
The second one works